src/rgen/Microsoft.Macios.Generator/DataModel/MethodReturnType.cs

```diff
@@ -0,0 +1,133 @@
+using System;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.Macios.Generator.Extensions;
+
+namespace Microsoft.Macios.Generator.DataModel;
+
+/// <summary>
+/// Readonly structure that represents a change in a method return type.
```

Comment:
I don't understand what is meant by "a change in"

Reply:
Read https://github.com/dotnet/roslyn/blob/main/docs/features/incremental-generators.cookbook.md

We get an event every time we get a code change. We calculate the data model and use that model to represent a code change. Use a comparer to invalidate the cache and move to generate if needed.

In this case, this is the data model to represent a change in the method return type.